src/NUnitFramework/framework/Internal/PropertyBag.cs

```diff
@@ -131,14 +142,14 @@ public TNode AddToXml(TNode parentNode, bool recursive)
         {
             TNode properties = parentNode.AddElement("properties");
 
-            foreach (string key in Keys)
+            foreach (var pair in inner)
```

Comment:
far better to enumerate dictionary directly than first getting keys and then getting value behind every key